src/main/java/net/minecraftforge/server/permission/context/IContext.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.server.permission.context;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+
+/**
+ * Use {@link PlayerContext} or {@link PlayerBlockContext} when possible
+ */
+public interface IContext
+{
+    /**
+     * World from where permission is requested. Can be null
+     */
+    @Nullable
+    World getWorld();
+
+    /**
+     * @return Player requesting permission. Can be null
+     */
+    @Nullable
+    EntityPlayer getPlayer();
```

Comment:
We could simplify it by having it all go through get, but yes performance is a issue on common ones, we need the game profile for the context, that will always exist, even if its Console's profile.
But world and entity are optional ones.